### ISSUE TITLE
test 11.7.2.4.2 for a PIV-I card should be testing for id-fpki-pivi-content-signing, not id-PIV-content-signing

### ISSUE THREAD

**Issue by dbcolston** [GSA/piv-conformance#251]
per "Worksheet 8: PIV-I Content Signing Certificate Profile" id-fpki-pivi-content-signing is "2.16.840.1.101.3.8.7"

**Comment by bob-fontana** [comment]
Can you tell us which database was being used? The issues opened have all of the markings of a PIV-I card being inserted, but the PIV database was selected.

**Comment by dbcolston** [comment]
I’ll retest tomorrow and let you know the database selected.  Do you want me to send you the config file used and/or test results?

Barry

**Comment by dbcolston** [comment]
PIV-I card #1:

 


2019-12-05 09:38:55

11.7.2.4.2

extended key usage (extKeyUsage) asserts id-PIV-content-signing.

Pass

Pass

 

            Extended key usage asserts 2.16.840.1.101.3.8.7, which is id-fpki-pivi-content-signing.  You may just need to change the description of the test to “extended key usage (extKeyUsage) asserts id-fpki-pivi-content-signing.”.

 

PIV-I card #2:

 

2019-12-05 09:51:14   11.7.2.4.2        extended key usage (extKeyUsage) asserts id-PIV-content-signing.            Pass     Pass

 

I can send you the signing certs if you want them

 

Barry

**Comment by bob-fontana** [comment]
Barry, so far so good, right?  Did you see any of the other failures?

**Comment by bob-fontana** [comment]
Please upload what you have from both runs (before and after).  That'll show whether the default database was running.  I've committed but not yet released a version that will force the user to select an appropriate database.  That should cut down on some of the false negatives.

**Comment by bob-fontana** [comment]
I can also see where some of the test case descriptions contain the NIST OID names for PIV, not PIV-I.  I'm opening a separate bug for that.

**Comment by bob-fontana** [comment]
I created a single bug to address these similar issues. The code is doing the right thing. It's the description of the test case that's incorrect.

**Comment by dbcolston** [comment]
Bob,

 

I don’t know what you are expecting for a “before” run.  Before what?

 

Barry

**Comment by dbcolston** [comment]
To whomever (maybe Bob Fontana):

 

I downloaded the FIPS-201-Card-Conformance-Tool-202003300729.zip file yesterday.  I ran the software against a PIV-I card and was expecting to see the 7 issues which I had previously created and which were all closed on February 27 and labelled as “Under Review”.  Only 1 of the issues has been resolved (#245).  I could find no way to change the status of the unresolved issues to Open, so I am sending this email.  For the tests, the PIV-I production card database (PIV-I_Production_Cards) was used and the card was a PIV-I card.  I have attached the debug log file.

 

The following is information about each of the 6 unresolved issues:

 

·       Issue #246 PIV-specific test 11.1.2.2.2 should not be executed for a PIV-I card – this test was executed, although it should not have executed for a PIV-I card.

 

2020-04-03 10:38:00,359 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:49] Test plan started for conformance test 11.1.2.2.2

2020-04-03 10:38:00,361 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.1.2.2.2:PKIX_X509DataObjectTests

2020-04-03 10:38:00,362 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.1.2.2.2:PKIX.6 test

2020-04-03 10:38:00,363 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.1.2.2.2:1 => oid = 2.16.840.1.101.3.7.2.1.1

 

2020-04-03 10:38:00,366 ERROR [Thread-4] g.g.c.testProgress [GuiTestListener.java:123] Test atom 11.1.2.2.2:1 => oid = 2.16.840.1.101.3.7.2.1.1 failed

2020-04-03 10:38:00,367 ERROR [Thread-4] g.g.p.GuiTestListener [GuiTestListener.java:131] Caused by:

java.lang.NullPointerException: null

            at gov.gsa.conformancelib.tests.PKIX_X509DataObjectTests.PKIX_Test_6(PKIX_X509DataObjectTests.java:249)

 

·       Issue #247 test 11.2.2.5 is testing for a PIV hardware OID instead of a PIV-I hardware OID.  Note: in addition to testing for the incorrect OID, the test description should be changed to specify a PIV-I OID, not a PIV OID

 

2020-04-03 10:38:03,267 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:49] Test plan started for conformance test 11.2.2.5

2020-04-03 10:38:03,272 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.2.2.5:PKIX_X509DataObjectTests

2020-04-03 10:38:03,275 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.2.2.5:PKIX.6 test

2020-04-03 10:38:03,282 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.2.2.5:1 => oid = 2.16.840.1.101.3.7.2.1.0

 

2020-04-03 10:38:03,288 ERROR [Thread-4] g.g.c.testProgress [GuiTestListener.java:123] Test atom 11.2.2.5:1 => oid = 2.16.840.1.101.3.7.2.1.0 failed

2020-04-03 10:38:03,288 ERROR [Thread-4] g.g.p.GuiTestListener [GuiTestListener.java:131] Caused by:

java.lang.NullPointerException: null

            at gov.gsa.conformancelib.tests.PKIX_X509DataObjectTests.PKIX_Test_6(PKIX_X509DataObjectTests.java:249)

 

·       Issue #248 test 11.3.2.4 is testing for a PIV hardware OID instead of a PIV-I hardware OID.  Note: in addition to testing for the incorrect OID, the test description should be changed to specify a PIV-I OID, not a PIV OID

 

2020-04-03 10:38:04,827 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:49] Test plan started for conformance test 11.3.2.4

2020-04-03 10:38:04,834 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.3.2.4:PKIX_X509DataObjectTests

2020-04-03 10:38:04,837 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.3.2.4:PKIX.6 test

2020-04-03 10:38:04,842 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.3.2.4:1 => oid = 2.16.840.1.101.3.7.2.1.2

 

2020-04-03 10:38:04,848 ERROR [Thread-4] g.g.c.testProgress [GuiTestListener.java:123] Test atom 11.3.2.4:1 => oid = 2.16.840.1.101.3.7.2.1.2 failed

2020-04-03 10:38:04,848 ERROR [Thread-4] g.g.p.GuiTestListener [GuiTestListener.java:131] Caused by:

java.lang.NullPointerException: null

            at gov.gsa.conformancelib.tests.PKIX_X509DataObjectTests.PKIX_Test_6(PKIX_X509DataObjectTests.java:249)

 

·       Issue #249 test 11.4.2.2 for a PIV-I card should be testing for id-fpki-certpcy-pivi-cardAuth, not id-fpki-common-cardAuth.  Note: the 2.16.840.1.101.3.7.2.5.0 OID being searched for in the Card Authentication Certificate is the OID for the Card Authentication Certificate container.  Note: in addition to testing for the incorrect OID, the test description should be changed to specify a PIV-I OID, not a PIV OID

 

2020-04-03 10:38:05,110 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:49] Test plan started for conformance test 11.4.2.2

2020-04-03 10:38:05,112 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.4.2.2:PKIX_X509DataObjectTests

2020-04-03 10:38:05,112 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.4.2.2:PKIX.6 test

2020-04-03 10:38:05,114 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.4.2.2:1 => oid = 2.16.840.1.101.3.7.2.5.0

 

2020-04-03 10:38:05,116 ERROR [Thread-4] g.g.c.testProgress [GuiTestListener.java:123] Test atom 11.4.2.2:1 => oid = 2.16.840.1.101.3.7.2.5.0 failed

2020-04-03 10:38:05,116 ERROR [Thread-4] g.g.p.GuiTestListener [GuiTestListener.java:131] Caused by:

java.lang.NullPointerException: null

            at gov.gsa.conformancelib.tests.PKIX_X509DataObjectTests.PKIX_Test_6(PKIX_X509DataObjectTests.java:249)

 

·       Issue #250 11.7.2.4.1 for a PIV-I card should be testing for id-fpki-certpcy-pivi-contentSigning, not id-fpki-common-piv-contentSigning.  Note: the 2.16.840.1.101.3.7.2.48.0 OID being searched for in the CHUID container is the OID for the CHUID container. 11.7.2.4.1.  Note: in addition to testing for the incorrect OID, the test description should be changed to specify a PIV-I OID, not a PIV OID

 

2020-04-03 10:38:06,741 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:49] Test plan started for conformance test 11.7.2.4.1

2020-04-03 10:38:06,743 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.7.2.4.1:PKIX_X509DataObjectTests

2020-04-03 10:38:06,743 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.7.2.4.1:PKIX.6 test

2020-04-03 10:38:06,745 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.7.2.4.1:1 => oid = 2.16.840.1.101.3.7.2.48.0

 

2020-04-03 10:38:06,749 ERROR [Thread-4] g.g.c.testProgress [GuiTestListener.java:123] Test atom 11.7.2.4.1:1 => oid = 2.16.840.1.101.3.7.2.48.0 failed

2020-04-03 10:38:06,749 ERROR [Thread-4] g.g.p.GuiTestListener [GuiTestListener.java:131] Caused by:

java.lang.NullPointerException: null

            at gov.gsa.conformancelib.tests.PKIX_X509DataObjectTests.PKIX_Test_6(PKIX_X509DataObjectTests.java:249)

 

·       Issue #251 test 11.7.2.4.2 for a PIV-I card should be testing for id-fpki-pivi-content-signing, not id-PIV-content-signing.  It is not clear what OID is being checked.  If it is the PIV-I OID that is being checked, then only the report description text needs to be changed to specify a PIV-I OID, not a PIV OID

 

2020-04-03 10:38:06,752 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:49] Test plan started for conformance test 11.7.2.4.2

2020-04-03 10:38:06,754 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.7.2.4.2:PKIX_X509DataObjectTests

2020-04-03 10:38:06,755 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.7.2.4.2:PKIX.20 test

2020-04-03 10:38:06,757 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:99] Started 11.7.2.4.2:1 => oid = 2.16.840.1.101.3.7.2.48.0

 

2020-04-03 10:38:06,760 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:133] Finished 11.7.2.4.2:1 => oid = 2.16.840.1.101.3.7.2.48.0

2020-04-03 10:38:06,761 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:133] Finished 11.7.2.4.2:PKIX.20 test

2020-04-03 10:38:06,761 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:133] Finished 11.7.2.4.2:PKIX_X509DataObjectTests

2020-04-03 10:38:06,761 INFO [Thread-4] g.g.c.testProgress [GuiTestListener.java:66] Test plan finished for conformance test 11.7.2.4.2

2020-04-03 10:38:06,762 INFO [Thread-4] g.g.c.testResult [GuiTestListener.java:67] 11.7.2.4.2,"extended key usage (extKeyUsage) asserts id-PIV-content-signing.",Pass,Pass

 

Barry

 

From: Barry Colston [mailto:bcolston@xtec.com] 
Sent: Thursday, December 05, 2019 9:54 AM
To: 'GSA/piv-conformance'; 'GSA/piv-conformance'
Cc: 'Author'
Subject: RE: [GSA/piv-conformance] test 11.7.2.4.2 for a PIV-I card should be testing for id-fpki-pivi-content-signing, not id-PIV-content-signing (#251)

 



 

PIV-I card #1:

 


2019-12-05 09:38:55

11.7.2.4.2

extended key usage (extKeyUsage) asserts id-PIV-content-signing.

Pass

Pass

 

            Extended key usage asserts 2.16.840.1.101.3.8.7, which is id-fpki-pivi-content-signing.  You may just need to change the description of the test to “extended key usage (extKeyUsage) asserts id-fpki-pivi-content-signing.”.

 

PIV-I card #2:

 

2019-12-05 09:51:14   11.7.2.4.2        extended key usage (extKeyUsage) asserts id-PIV-content-signing.            Pass     Pass

 

I can send you the signing certs if you want them

 

Barry
